src/vmm/src/gdb/hw_breakpoint_store.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use vm_memory::GuestAddress;
+
+/// Used to store hardware breakpoints for GDB debugging, it prevents the possiblity of storing
+/// more breakpoints than the allocated size
+#[derive(Debug)]
+pub struct HwBpStore<const S: usize> {
```

Comment:
Can we just use an `ArrayVec` here (https://docs.rs/arrayvec/latest/arrayvec/struct.ArrayVec.html)? It'd be a new dependency, but I'd probably be in favor of adding it over implementing our own data structures.

Reply:
Updated with the arrayvec, made it only load with the GDB flag as well